File: frontend/src/services/userService.ts
import axios from 'axios';
import { User } from '../types';
import { PageResponse } from '../types';

const API_URL = process.env.REACT_APP_API_URL || 'http://localhost:8080/api';

export const userService = {
    getAllUsers: async (page: number = 0, size: number = 10, sortBy: string = 'username', direction: string = 'asc'): Promise<PageResponse<User>> => {
        const response = await axios.get<PageResponse<User>>(`${API_URL}/users`, {
            params: { page, size, sortBy, direction }
        });
        return response.data;
    },

    searchUsers: async (searchTerm: string, page: number = 0, size: number = 10, sortBy: string = 'username', direction: string = 'asc'): Promise<PageResponse<User>> => {
        const response = await axios.get<PageResponse<User>>(`${API_URL}/users/search`, {
            params: { searchTerm, page, size, sortBy, direction }
        });
        return response.data;
    },

    getUsersByStatus: async (enabled: boolean, page: number = 0, size: number = 10, sortBy: string = 'username', direction: string = 'asc'): Promise<PageResponse<User>> => {
        const response = await axios.get<PageResponse<User>>(`${API_URL}/users/status`, {
            params: { enabled, page, size, sortBy, direction }
        });
        return response.data;
    },

    getLockedUsers: async (page: number = 0, size: number = 10, sortBy: string = 'username', direction: string = 'asc'): Promise<PageResponse<User>> => {
        const response = await axios.get<PageResponse<User>>(`${API_URL}/users/locked`, {
            params: { page, size, sortBy, direction }
        });
        return response.data;
    },

    createUser: async (user: Omit<User, 'id'>): Promise<User> => {
        const response = await axios.post<User>(`${API_URL}/users`, user);
        return response.data;
    },

    updateUser: async (id: number, user: Partial<User>): Promise<User> => {
        const response = await axios.put<User>(`${API_URL}/users/${id}`, user);
        return response.data;
    },

    deleteUser: async (id: number): Promise<void> => {
        await axios.delete(`${API_URL}/users/${id}`);
    },

    changePassword: async (id: number, oldPassword: string, newPassword: string): Promise<void> => {
        await axios.post(`${API_URL}/users/${id}/change-password`, null, {
            params: { oldPassword, newPassword }
        });
    },

    resetPassword: async (email: string): Promise<void> => {
        await axios.post(`${API_URL}/users/reset-password`, null, {
            params: { email }
        });
    },

    unlockAccount: async (id: number): Promise<void> => {
        await axios.post(`${API_URL}/users/${id}/unlock`);
    }
}; 

File: backend/src/main/java/com/xml/processor/service/UserService.java
package com.xml.processor.service;

import com.xml.processor.model.User;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.Optional;

public interface UserService {
    User createUser(User user);
    User updateUser(Long id, User user);
    void deleteUser(Long id);
    Optional<User> getUserById(Long id);
    Optional<User> getUserByUsername(String username);
    Optional<User> getUserByEmail(String email);
    Page<User> getAllUsers(Pageable pageable);
    Page<User> searchUsers(String searchTerm, Pageable pageable);
    Page<User> getUsersByStatus(boolean enabled, Pageable pageable);
    Page<User> getLockedUsers(Pageable pageable);
    void changePassword(Long userId, String oldPassword, String newPassword);
    void resetPassword(String email);
    void unlockAccount(Long userId);
    void incrementFailedLoginAttempts(String username);
    void resetFailedLoginAttempts(String username);
    boolean isAccountLocked(String username);
} 